azurerm/data_source_storage_account_sas.go

```diff
@@ -0,0 +1,390 @@
+package azurerm
+
+import "github.com/hashicorp/terraform/helper/schema"
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+)
+
+const (
+	connStringAccountKeyKey  = "AccountKey"
+	connStringAccountNameKey = "AccountName"
+	sasSignedVersion         = "2017-07-29"
+)
+
+// This is an ACCOUNT SAS : https://docs.microsoft.com/en-us/rest/api/storageservices/Constructing-an-Account-SAS
+// not Service SAS
+func dataSourceArmStorageAccountSharedAccessSignature() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmStorageAccountSasRead,
+
+		Schema: map[string]*schema.Schema{
+			"connection_string": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+
+			"https_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+				ForceNew: true,
+			},
+
+			"resource_types": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"service": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"container": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"object": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"services": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"blob": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"queue": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"table": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+
+						"file": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			// Always in UTC and must be ISO-8601 format
+			"start": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			// Always in UTC and must be ISO-8601 format
+			"expiry": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
I'd prefer to go the route you specify with the validation function. In my research there are some slight differences between ISO-8601 and RFC3339. MSFT specifies that these are ISO-8601 and since the signatures are based on the actual text characters, the subtle differences may result in breakage/no-working. 

Given the fragile nature, by design, of the signatures, I think leaving as-is is preferable rather than constraining the input to possibly broken validation.

Reply:
makes sense from my side - we can always add it in later based on feedback 👍